### PR TITLE
Use `tool.scikit-build.cmake.version`

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Description
`cmake.minimum-version` has been deprecated since `scikit-build-core` 0.8, and is now causing conflicts in 0.10 due to its attempts to auto-detect `cmake.version` from `CMakeLists.txt`. Bump the minimum `scikit-build-core` to 0.10 and use the suggested `cmake.version`.

Contributes to https://github.com/rapidsai/build-planning/issues/58

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/librmm/pyproject.toml [hunk 1]
```diff
@@ -55,7 +55,7 @@ requires = [
 [tool.scikit-build]
 build-dir = "build/{wheel_tag}"
 cmake.build-type = "Release"
-cmake.minimum-version = "3.26.4"
+cmake.version = "CMakeLists.txt"
```
Comment (bdice) [inline python/librmm/pyproject.toml:58]: Great. This PR tackles (1) and (2) from https://github.com/rapidsai/build-planning/issues/58. Should we do (3) as well?

> set a minimum-version (minimum scikit-build-core version) in the [tool.scikit-build-core] table in pyproject.toml

```suggestion
cmake.version = "CMakeLists.txt"
minimum-version = "0.10"
```

**Review by bdice — APPROVED** [review]
Just one suggestion, otherwise LGTM. Let's apply these changes across all of RAPIDS.

Should we be concerned about this affecting the 24.08 release?

File: python/librmm/pyproject.toml (hunk 1 above)
Comment (jameslamb) [reply python/librmm/pyproject.toml:58]: +1, was just about to say the same thing. If we're going to have to touch all these configs, might as well also close https://github.com/rapidsai/build-planning/issues/58 in the process.

**Comment by henryiii** [comment]
> causing conflicts in 0.10 due to its attempts to auto-detect cmake.version from CMakeLists.txt

It's not supposed to do this unless minimum-version is unset or set to 0.10+.

**Comment by henryiii** [comment]
Oh, is `minimum-version` unset and you are using `cmake.minimum-version`? I might not have worked out that possibility. What's breaking, though, if it auto-detects the version?

**Comment by bdice** [comment]
https://github.com/rapidsai/rmm/pull/1638#issuecomment-2272245102

That's right, @henryiii. scikit-build-core 0.10.0 sees that the `minimum-version` (for scikit-build-core) is unset, and thus sets `cmake.version`, which conflicts with our manually set `cmake.minimum-version`.

**Comment by henryiii** [comment]
But it's not supposed to do this. I'm trying to see if I can reproduce this in our tests. It's supposed to take minimum-version if it's set and version isn't.

**Comment by henryiii** [comment]
I see the bug. I don't see why I can't trigger the bug in our tests yet. Could you try a branch? https://github.com/scikit-build/scikit-build-core/pull/853

**Comment by jameslamb** [comment]
> Could you try a branch?

Sure! I just pushed to https://github.com/rapidsai/rmm/pull/1638.

The `wheel-build-*` jobs will be the ones to watch: https://github.com/rapidsai/rmm/actions/runs/10275261685/job/28433610205?pr=1638

Using that PR which contains 0 other changes instead of this one because it'd tell us if new behavior from your `scikit-build-core` branch would allow RAPIDS to build successfully with a new version of `scikit-build-core` that contains that patch.

**Comment by jameslamb** [comment]
@henryiii it does look to me like the wheel builds are succeeding with your patch!

https://github.com/rapidsai/rmm/actions/runs/10275391215/job/28434079051?pr=1638

*(ignore the failing `conda-*` jobs there... those are still pulling in `scikit-build-core==0.10.0`)*

**Comment by jakirkham** [comment]
We are seeing this error [on CI]( https://github.com/rapidsai/rmm/actions/runs/10274129734/job/28434909785?pr=1637#step:7:1664 ):

```python
    ERROR: Cannot set cmake.verbose if minimum-version is set to 0.10 or higher
```

This should be fixed by upstream PR: https://github.com/rapidsai/devcontainers/pull/377

**Comment by henryiii** [comment]
0.10.1 is out, with this regression fixed. In general, not setting `minimum-version` means you might be affected by updates, but this was a mistake. I'd recommend going 0.10+, of course, but the regression should be fixed. :)

**Comment by jameslamb** [comment]
Thank you SO MUCH for the quick fix @henryiii !!!

You're right of course, we should be setting `minimum-version`.

**Comment by henryiii** [comment]
Pushed to conda-forge too, that should be live in a few minutes.

**Comment by jameslamb** [comment]
The `devcontainers` build should be fixed once https://github.com/rapidsai/devcontainers/pull/377 is merged.

File: python/librmm/pyproject.toml [hunk 2]
```diff
@@ -55,7 +55,8 @@ requires = [
...
+minimum-version = "0.10"
```
Comment (jakirkham) [inline python/librmm/pyproject.toml:59]: As pointed out by Henry ( https://github.com/rapidsai/build-planning/issues/58#issuecomment-2274214167 ), we can configure `scikit-build-core` to retrieve the minimum version from `build-system.requires`

This is also mentioned in:

* [This blogpost]( https://iscinumpy.dev/post/scikit-build-core-0-10/ )
* [The scikit-build-core docs]( https://scikit-build-core.readthedocs.io/en/latest/configuration.html#minimum-version-defaults )

Here is what this change would entail:

```suggestion
minimum-version = "build-system.requires"
```

We probably want to roll this out across the existing PRs

cc @jameslamb (for awareness)

Comment (jameslamb) [reply python/librmm/pyproject.toml:59]: Agreed! I intentionally hadn't included this in these PRs last night, when we were confronting the possibility of them being last-minute changes for the 24.08 release. Wanted to change as little as was necessary to get things working again.

Now that that's not the case, I totally support this. Really nice feature 🤩 .

I'll go apply this across the other PRs if it works here in `rmm`.

File: python/rmm/pyproject.toml [hunk 3]
```diff
@@ -110,7 +110,8 @@ skip = [
 [tool.scikit-build]
 build-dir = "build/{wheel_tag}"
 cmake.build-type = "Release"
-cmake.minimum-version = "3.26.4"
+cmake.version = "CMakeLists.txt"
+minimum-version = "0.10"
```
Comment (jameslamb) [inline python/rmm/pyproject.toml:114]: minimum-version = "build-system.requires"

Comment (jameslamb) [inline python/rmm/pyproject.toml:114]:
```suggestion
minimum-version = "build-system.requires"
```

File: python/librmm/pyproject.toml (hunk 2 above)
Comment (jameslamb) [reply python/librmm/pyproject.toml]: Looks like this worked well! I've updated all the other PRs from https://github.com/rapidsai/build-planning/issues/58

Comment (jakirkham) [reply python/librmm/pyproject.toml]: Thanks James! 🙏

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

**Pull request merged**